app/utils/deep_link/index.ts

```diff
@@ -114,37 +120,66 @@ export async function handleDeepLink(deepLinkUrl: string, intlShape?: IntlShape,
     }
 }
 
+const matcherOpts: Parameters<typeof match>[1] = {decode: decodeURIComponent};
+
+type ChannelPathParams = {
+    hostname: string;
+    subpath?: string;
+    teamName: string;
+    path: 'channels' | 'messages';
+    identifier: string;
+    postId?: string;
+};
+
+const CHANNEL_PATH = `/:subpath*/:teamName(${TEAM_NAME_PATH_PATTERN})/:path(channels|messages)/:identifier(${IDENTIFIER_PATH_PATTERN})/:postId(${ID_PATH_PATTERN})?`;
```

Comment:
If you have a channel named `messages` (which doesn't sound unreasonable), a deeplink to a post in that channel ends up looking like this:

```
/teamname/channels/messages/12345678901234567890123456 
```

This gets parsed incorrectly, because `teamname` gets interpreted as the subpath and `channels` as the team name.

Reply:
refactored/fixed, host and pathname are handled in the same `:serverUrl` parameter now, that was previously a workaround for the deep subpath issue, but obviously caused other issues.